search_spaces_blackbox_test.go

```diff
@@ -0,0 +1,176 @@
+	"github.com/almighty/almighty-core/app/test"
+	"github.com/almighty/almighty-core/application"
+	"github.com/almighty/almighty-core/gormapplication"
+	"github.com/almighty/almighty-core/resource"
+	"github.com/almighty/almighty-core/space"
+	"github.com/goadesign/goa"
+	"golang.org/x/net/context"
+	"strings"
+)
+
+type args struct {
+	pageOffset *string
+	pageLimit  *int
+	q          string
+}
+
+type expect func(*testing.T, okScenario, *app.SearchSpaceList)
+type expects []expect
+
+type okScenario struct {
+	name    string
+	args    args
+	expects expects
+}
+
+func TestSpacesSearchOK(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	idents := createTestData()
+	defer cleanTestData(idents)
+
+	tests := []okScenario{
+		{"With uppercase fullname query", args{offset("0"), limit(10), "TEST_AB"}, expects{totalCount(1)}},
+		{"With lowercase fullname query", args{offset("0"), limit(10), "TEST_AB"}, expects{totalCount(1)}},
+		{"With uppercase description query", args{offset("0"), limit(10), "DESCRIPTION FOR TEST_AB"}, expects{totalCount(1)}},
+		{"With lowercase description query", args{offset("0"), limit(10), "description for test_ab"}, expects{totalCount(1)}},
+		{"with special chars", args{offset("0"), limit(10), "&:\n!#%?*"}, expects{totalCount(0)}},
+		{"with * to list all", args{offset("0"), limit(10), "*"}, expects{totalCountAtLeast(len(idents))}},
+		{"with multi page", args{offset("0"), limit(10), "TEST"}, expects{hasLinks("Next")}},
+		{"with last page", args{offset(strconv.Itoa(len(idents) - 1)), limit(10), "TEST"}, expects{hasNoLinks("Next"), hasLinks("Prev")}},
+		{"with different values", args{offset("0"), limit(10), "TEST"}, expects{differentValues()}},
+	}
+
+	service := goa.New("TestSpacesSearch-Service")
+	controller := NewSearchController(service, gormapplication.NewGormDB(DB))
+
+	for _, tt := range tests {
+		_, result := test.SpacesSearchOK(t, context.Background(), service, controller, tt.args.pageLimit, tt.args.pageOffset, tt.args.q)
+		for _, expect := range tt.expects {
+			expect(t, tt, result)
+		}
+	}
+}
+
+func createTestData() []space.Space {
+	names := []string{"TEST_A", "TEST_AB", "TEST_B", "TEST_C"}
+	for i := 0; i < 20; i++ {
+		names = append(names, "TEST_"+strconv.Itoa(i))
+	}
+
+	spaces := []space.Space{}
+
+	err := application.Transactional(gormapplication.NewGormDB(DB), func(app application.Application) error {
+		for _, name := range names {
+			space := space.Space{
+				Name: name,
+				Description: strings.ToTitle("description for " + name),
+			}
+			newSpace, err := app.Spaces().Create(context.Background(), &space)
+			if err != nil {
+				return err
+			}
+			spaces = append(spaces, *newSpace)
+		}
+		return nil
+	})
+	if err != nil {
+		fmt.Println("Failed to insert testdata", err)
+	}
+	return spaces
+}
+
+func cleanTestData(spaces []space.Space) {
+	err := application.Transactional(gormapplication.NewGormDB(DB), func(app application.Application) error {
```

Comment:
Use DeletedCreatedEntities instead of manually cleaning up: 
https://github.com/almighty/almighty-core/blob/master/work-item-comments_test.go#L45